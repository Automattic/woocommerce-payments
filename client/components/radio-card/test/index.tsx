/**
 * External dependencies
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';

/**
 * Internal dependencies
 */
import RadioCard from '../';

const options = [
	{
		label: 'Pineapple pizza',
		value: 'pineapple',
		icon: <svg />,
		content: <p>Sweet pizza</p>,
	},
	{
		label: 'Pizza',
		value: 'pizza',
		icon: <svg />,
		content: <p>The real pizza</p>,
	},
];

describe( 'RadioCard Component', () => {
	it( 'renders RadioCard component with provided props', () => {
		const { container } = render(
			<RadioCard
				name="pizzas"
				selected="pineapple"
				onChange={ jest.fn() }
				options={ options }
			/>
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'changes the selected value when an option is clicked', () => {
		const mockOnChange = jest.fn();
		render(
			<RadioCard
				name="pizzas"
				selected="pizza"
				onChange={ mockOnChange }
				options={ options }
			/>
		);

		user.click( screen.getByLabelText( /Pineapple/i ) );
		expect( mockOnChange ).toHaveBeenCalledWith( 'pineapple' );
	} );
} );
