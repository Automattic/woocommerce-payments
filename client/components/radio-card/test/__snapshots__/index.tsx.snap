// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioCard Component renders RadioCard component with provided props 1`] = `
<div>
  <label
    class="wcpay-component-radio-card checked"
  >
    <div
      class="wcpay-component-radio-card__label"
    >
      <svg />
      Pineapple pizza
      <input
        checked=""
        name="pizzas"
        type="radio"
        value="pineapple"
      />
    </div>
    <p>
      Sweet pizza
    </p>
  </label>
  <label
    class="wcpay-component-radio-card"
  >
    <div
      class="wcpay-component-radio-card__label"
    >
      <svg />
      Pizza
      <input
        name="pizzas"
        type="radio"
        value="pizza"
      />
    </div>
  </label>
</div>
`;
